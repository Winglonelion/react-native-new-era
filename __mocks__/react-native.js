import { Animated as RNAnimated } from 'react-native';

export const Animated = {
  ...RNAnimated,
  parallel: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
  timing: () => ({
    // immediately invoke callback
    start: cb => cb(),
  }),
};
