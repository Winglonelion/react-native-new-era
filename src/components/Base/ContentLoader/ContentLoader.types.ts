import { SvgProps } from 'react-native-svg';

export interface IContentLoaderProps extends SvgProps {
  animate?: boolean;
  backgroundColor?: string;
  foregroundColor?: string;
  rtl?: boolean;
  speed?: number;
  interval?: number;
  uniqueKey?: string;
}
