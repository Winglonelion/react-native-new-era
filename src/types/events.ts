import { GestureResponderEvent } from 'react-native';

export type PressEvent = (event: GestureResponderEvent) => any;
