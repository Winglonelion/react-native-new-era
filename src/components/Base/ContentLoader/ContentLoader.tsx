import * as React from 'react';
import { Circle, Path, Rect } from 'react-native-svg';

import { IContentLoaderProps } from './ContentLoader.types';
import ReactContentLoaderFacebook from './presets/FacebookStyle';
import NativeSvg from './Svg';

const ContentLoader: React.FC<IContentLoaderProps> = props =>
  props.children ? (
    <NativeSvg {...props} />
  ) : (
    <ReactContentLoaderFacebook {...props} />
  );

export { Circle, Rect, Path };

export default ContentLoader;
