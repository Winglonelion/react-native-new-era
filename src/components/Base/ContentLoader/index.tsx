import ContentLoader from './ContentLoader';

export { Circle, Rect, Path } from './ContentLoader';
export default ContentLoader;
