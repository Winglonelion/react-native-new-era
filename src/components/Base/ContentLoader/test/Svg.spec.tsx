import React from 'react';
import { shallow } from 'enzyme';
import Svg from '../Svg';
import { IContentLoaderProps } from '../ContentLoader.types';
import { View, Animated as RNAnimated } from 'react-native';

export const Animated = {
  ...RNAnimated,
  parallel: () => ({
    start: (cb: any) => cb(),
  }),
  // Animated.timing
  timing: (value: any, config: any) => {
    return {
      start: (callback: any) => {
        value.setValue(config.toValue);
        callback && callback();
      },
    };
  },
};

const setup = (
  propOverrides: IContentLoaderProps = {} as IContentLoaderProps,
) => {
  const props = { ...propOverrides };
  const wrapper = shallow(
    <Svg {...props}>
      <View />
    </Svg>,
  );
  return {
    props,
    wrapper,
  };
};

describe('Svg check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup({ uniqueKey: 'xxxxxx' });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Svg check snapshot with props', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup({
      uniqueKey: 'xxxxxx',
      animate: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
