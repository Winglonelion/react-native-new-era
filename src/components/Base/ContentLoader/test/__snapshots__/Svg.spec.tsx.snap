// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg check snapshot should match snapshot 1`] = `
<Svg
  interval={0.25}
  preserveAspectRatio="xMidYMid meet"
  style={Object {}}
>
  <Rect
    clipPath="url(#xxxxxx-animated-diff)"
    fill="url(#xxxxxx-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="xxxxxx-animated-diff"
    >
      <View />
    </ClipPath>
    <ForwardRef(AnimatedComponentWrapper)
      id="xxxxxx-diff"
      x1={"-100%"}
      x2={"0%"}
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0.5}
        stopColor="#eee"
      />
      <Stop
        offset={1}
        stopColor="#f5f6f7"
      />
    </ForwardRef(AnimatedComponentWrapper)>
  </Defs>
</Svg>
`;

exports[`Svg check snapshot with props should match snapshot 1`] = `
<Svg
  interval={0.25}
  preserveAspectRatio="xMidYMid meet"
  style={Object {}}
>
  <Rect
    clipPath="url(#xxxxxx-animated-diff)"
    fill="url(#xxxxxx-diff)"
    height="100%"
    width="100%"
    x="0"
    y="0"
  />
  <Defs>
    <ClipPath
      id="xxxxxx-animated-diff"
    >
      <View />
    </ClipPath>
    <ForwardRef(AnimatedComponentWrapper)
      id="xxxxxx-diff"
      x1={"-100%"}
      x2={"0%"}
      y1={0}
      y2={0}
    >
      <Stop
        offset={0}
        stopColor="#f5f6f7"
      />
      <Stop
        offset={0.5}
        stopColor="#eee"
      />
      <Stop
        offset={1}
        stopColor="#f5f6f7"
      />
    </ForwardRef(AnimatedComponentWrapper)>
  </Defs>
</Svg>
`;
